control-plane/api/v1alpha1/registration_webhook_test.go

```diff
@@ -0,0 +1,330 @@
+					Name: "test-registration",
+				},
+				Spec: RegistrationSpec{
+					Node:        "",
+					Address:     "10.2.2.1",
+					Service:     Service{Name: "test-service"},
+					HealthCheck: nil,
+				},
+			},
+			expAllow:      false,
+			expErrMessage: "registration.Spec.Node is required",
+		},
+		"invalid, missing address field": {
+			newResource: &Registration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-registration",
+				},
+				Spec: RegistrationSpec{
+					Node:        "test-node",
+					Address:     "",
+					Service:     Service{Name: "test-service"},
+					HealthCheck: nil,
+				},
+			},
+			expAllow:      false,
+			expErrMessage: "registration.Spec.Address is required",
+		},
+		"invalid, missing service.name field": {
+			newResource: &Registration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-registration",
+				},
+				Spec: RegistrationSpec{
+					Node:        "test-node",
+					Address:     "10.2.2.1",
+					Service:     Service{Name: ""},
+					HealthCheck: nil,
+				},
+			},
+			expAllow:      false,
+			expErrMessage: "registration.Spec.Service.Name is required",
+		},
+		"invalid, health check is set and name is missing": {
+			newResource: &Registration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-registration",
+				},
+				Spec: RegistrationSpec{
+					Node:    "test-node",
+					Address: "10.2.2.1",
+					Service: Service{Name: "test-service"},
+					HealthCheck: &HealthCheck{
+						Name:   "",
+						Status: "passing",
+						Definition: HealthCheckDefinition{
+							IntervalDuration: "10s",
+						},
+					},
+				},
+			},
+			expAllow:      false,
+			expErrMessage: "registration.Spec.HealthCheck.Name is required",
+		},
+		"invalid, health check is set and intervalDuration is missing": {
+			newResource: &Registration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-registration",
+				},
+				Spec: RegistrationSpec{
+					Node:    "test-node",
+					Address: "10.2.2.1",
+					Service: Service{Name: "test-service"},
+					HealthCheck: &HealthCheck{
+						Name:   "check name",
+						Status: "passing",
+						Definition: HealthCheckDefinition{
+							IntervalDuration: "",
+						},
+					},
+				},
+			},
+			expAllow:      false,
+			expErrMessage: "invalid registration.Spec.HealthCheck.Definition.IntervalDuration value: \"\"",
+		},
+		"invalid, health check is set and intervalDuration is invalid duration type": {
+			newResource: &Registration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-registration",
+				},
+				Spec: RegistrationSpec{
+					Node:    "test-node",
+					Address: "10.2.2.1",
+					Service: Service{Name: "test-service"},
+					HealthCheck: &HealthCheck{
+						Name:   "check name",
+						Status: "passing",
+						Definition: HealthCheckDefinition{
+							IntervalDuration: "150",
+						},
+					},
+				},
+			},
+			expAllow:      false,
+			expErrMessage: "invalid registration.Spec.HealthCheck.Definition.IntervalDuration value: \"150\"",
```

Comment:
I'm wondering if this should be a more descriptive error message. Are we expecting int only for duration in seconds?

Reply:
so it needs to be something like `5m` or `30s` (from our docs https://developer.hashicorp.com/consul/docs/services/configuration/checks-configuration-reference)